### PR TITLE
Tests for DIS-41

### PR DESCRIPTION
**Pull request by jogura**
Initial checkin for DIS-81 Tests for DIS-41 Display ThreadList
* Add additional tests for getting specific forums with threads
* Add methods to convert HalResponses for Threads, Posts, and PostRevisions
* Add SecureActivity for VIEW_FORUM (stubbed impl ATM)
* Refactor ForumResourceTest to use Hamcrest assertThat
* Introduce new generic Hamcrest Matcher for Content comparisons by id from a list
* Refactor all Mappers to not stringify Integer types (we are only stringifying Long)

### CODE REVIEW

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceIntegrationTest.java [hunk 1]
```diff
@@ -85,19 +88,54 @@ public static void tearDownClass() {
...
+    Forum rootForum = getRootForum(SITE_ID_GLEE);
+    Long firstForumId = rootForum.getChildren().get(0).getId();
+
+    Forum firstForum = getForum(SITE_ID_GLEE, firstForumId);
+    assertNotNull(firstForum);
+    assertNotNull(firstForum.getThreads());
+    assertTrue("threads not empty", !firstForum.getThreads().isEmpty());
+    ForumThread firstThread = firstForum.getThreads().get(0);
+    assertNotNull(firstThread);
+    assertNotNull(firstThread.getFirstPost());
```
Comment (armonr) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceIntegrationTest.java]: Using `instanceof` makes these checks more robust

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceTest.java [hunk 2]
```diff
@@ -103,8 +112,41 @@ public void setup() {
     return forumList;
   }
 
+  private List<ForumThread> createForumThreads(Integer siteId, Long forumId, int limit) {
+    List<ForumThread> forumThreadList = new ArrayList<>();
+
+    LongStream.range(1, 1 + limit).forEach(
+        i -> {
+        Long threadId = i + 1000L;
```
Comment (armonr) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceTest.java]: Minor: should `Long` operator be used?

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceTest.java [hunk 3]
```diff
@@ -119,28 +161,28 @@ public void testDefaultGetForums() throws IOException {
     when(forumService.getRootForum(1)).thenReturn(optionalForum);
 
     Response response = RULE.client().target("/1/forums").request().get();
-    assertThat(response.getStatus()).isEqualTo(200);
+    assertThat(response.getStatus(), equalTo(200));
```
Comment (armonr) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceTest.java]: Why is this version preferable?

Comment (jogura) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceTest.java]: To keep everything consistent. Below I added a custom Hamcrest matcher to help with asserting that collections of Content existing based on their IDs.

File: service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java [hunk 4]
```diff
@@ -25,34 +28,22 @@
   public void getForumRootWithNestedChildren() {
     List<Forum> forums = new ArrayList<>();
 
-    Forum rootForum = new Forum();
-    rootForum.setId(Forum.FORUM_ROOT_ID);
-    rootForum.setParentId(Forum.FORUM_ROOT_ID);
+    Forum rootForum = createForum(Forum.FORUM_ROOT_ID, Forum.FORUM_ROOT_ID);
```
Comment (armonr) [inline service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java]: First argument is site id. maybe a different constant can be made for this?

File: service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java [hunk 5]
```diff
@@ -100,4 +91,78 @@ public void getForumRootWithNestedChildren() {
 
     verify(forumDao).getForumList(anyInt());
   }
+
+  private Forum createForum(Long id, Long parentId) {
+    return createForum(id, parentId, String.format("Forum: %s", id));
+  }
+
+  private Forum createForum(Long id, Long parentId, String name) {
```
Comment (armonr) [inline service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java]: site id should be `int`

File: service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java (hunk 4 above)
Comment (jogura) [inline service/discussion/src/test/java/com/wikia/discussionservice/services/ForumServiceTest.java]: The first argument is the forum's `id` the second is `parentId`

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceIntegrationTest.java (hunk 1 above)
Comment (jogura) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/ForumResourceIntegrationTest.java]: Good suggestion: a30aad1

**Comment by jimmordino** [comment]
Other than the things we discussed, LGTM.  At some point we should refactor all of our tests to use ```Response.Status``` constants for HTTP return codes instead of numeric constants.

:+1:

**Comment by armonr** [comment]
LGTM too

**Pull request merged**